app/models/ext_management_system.rb

```diff
@@ -131,6 +140,20 @@ def self.provider_create_params
   supports :refresh_ems
   supports_not :assume_role
 
+  def edit_with_params(params, endpoints, authentications)
+    tap do |ems|
+      # Remove endpoints/attributes that are not arriving in the arguments above
+      ems.endpoints.where.not(:role => nil).where.not(:role => endpoints.map { |ep| ep['role'] }).delete_all
+      ems.authentications.where.not(:authtype => nil).where.not(:authtype => authentications.map { |au| au['authtype'] }).delete_all
```

Comment:
Since these include `.delete_all`'s , I wonder if we should wrap this whole thing in a transaction to avoid 2 API calls colliding with each other.

Reply:
I guess probably the same comment above with the create_from_params.

Reply:
Makes sense, will do that.